typescript/cli/src/tests/ethereum/warp/warp-deploy.e2e-test.ts

```diff
@@ -72,6 +74,32 @@ const WARP_CORE_CONFIG_PATH_2_3 = GET_WARP_DEPLOY_CORE_CONFIG_OUTPUT_PATH(
   'VAULT',
 );
 
+function extractInputOnlyFields(config: TokenFeeConfigInput): any {
+  if (!config) return config;
+
+  switch (config.type) {
+    case TokenFeeType.LinearFee:
+      return {
+        type: config.type,
+        bps: config.bps.toString(), // Convert to string for consistent comparison
+      };
+    case TokenFeeType.RoutingFee:
+      return {
+        type: config.type,
+        ...(config.feeContracts && {
+          feeContracts: objMap(config.feeContracts, (_, subConfig) =>
+            extractInputOnlyFields(subConfig),
+          ),
+        }),
+      };
+    case TokenFeeType.ProgressiveFee:
+    case TokenFeeType.RegressiveFee:
+      return _.pick(config, ['type']);
```

Comment:
i think that you could have used the `pick` utils function from the utils package
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/utils/src/objects.ts#L91